app/src/main/java/com/andrew/tasky/presentation/register/RegisterFragment.kt

```diff
@@ -28,5 +31,14 @@ class RegisterFragment : Fragment(R.layout.fragment_register) {
         binding.backButton.setOnClickListener(){
             navController.popBackStack()
         }
+
+        binding.nameEditText.addTextChangedListener{
+            if (it != null) {
+                binding.nameCheckBox.isVisible = it.length in 2..50
```

Comment:
This will not survive config changes. That's UI state that belongs in the ViewModel. For example, a StateFlow called `isNameValid`